unit-tests/py/rspy/file.py

```diff
@@ -0,0 +1,124 @@
+
+import os, re, platform, subprocess, sys
+
+# Remove Python's default list of places to look for modules so that we use our log module
+sys.path = list()
+sys.path.append( '' )  # directs Python to search modules in the current directory first
+sys.path.append( os.path.dirname( sys.executable ))
+sys.path.append( os.path.join( os.path.dirname( sys.executable ), 'DLLs' ))
+sys.path.append( os.path.join( os.path.dirname( sys.executable ), 'lib' ))
+# Add our py/ module directory
+current_dir = os.path.dirname( os.path.abspath( __file__ ))
+sys.path.append( os.path.dirname( current_dir ) )
+
+from rspy import log
+
+# get os and directories for future use
+# NOTE: WSL will read as 'Linux' but the build is Windows-based!
+system = platform.system()
+if system == 'Linux'  and  "microsoft" not in platform.uname()[3].lower():
+    linux = True
+else:
+    linux = False
+
+def filesin( root ):
+    """
+    Yield all files found in root, using relative names ('root/a' would be yielded as 'a')
+    """
+    for (path,subdirs,leafs) in os.walk( root ):
+        for leaf in leafs:
+            # We have to stick to Unix conventions because CMake on Windows is fubar...
+            yield os.path.relpath( path + '/' + leaf, root ).replace( '\\', '/' )
+
+def find( dir, mask ):
+    """
+    Yield all files in given directory (including sub-directories) that fit the given mask
+    :param dir: directory in which to search
+    :param mask: mask to compare file names to
+    """
+    pattern = re.compile( mask )
+    for leaf in filesin( dir ):
+        if pattern.search( leaf ):
+            #log.d(leaf)
+            yield leaf
+
+def is_executable(path_to_file):
+    """
+    :param path_to_file: path to a file
+    :return: whether the file is an executable or not
+    """
+    global linux
+    if linux:
+        return os.access(path_to_file, os.X_OK)
+    else:
+        return path_to_file.endswith('.exe')
+
+def subprocess_run(cmd, stdout = None):
+    """
+    wrapper function for subprocess.run
+    """
+    log.d( 'running:', cmd )
+    handle = None
+    try:
+        log.debug_indent()
+        if stdout  and  stdout != subprocess.PIPE:
+            handle = open( stdout, "w" )
+            stdout = handle
+        rv = subprocess.run( cmd,
+                             stdout = stdout,
+                             stderr = subprocess.STDOUT,
+                             universal_newlines = True,
+                             check = True)
+        result = rv.stdout
+        if not result:
+            result = []
+        else:
+            result = result.split( '\n' )
+        return result
+    finally:
+        if handle:
+            handle.close()
+        log.debug_unindent()
+
+def remove_newlines (lines):
+    for line in lines:
+        if line[-1] == '\n':
+            line = line[:-1]    # excluding the endline
+        yield line
+
+def grep_( pattern, lines, context ):
```

Comment:
How about we rename this `_grep` so it won't be exposed?

Reply:
as you wish